File: grading_students.py
# if the difference in the grade and the next multiple of 5 is less than three round up.
# if under 38 no rounding.
# output given sample is 75 67 40 33
import math

grades = [73, 67, 38, 33]


def grading_students(grades):
    final_grades = []
    for grade in grades:
        if grade <= 37:
            final_grades.append(grade)
        elif (grade + 2) % 5 == 0:
            final_grades.append(round(grade / 5) * 5)
        elif (grade + 1) % 5 == 0:
            final_grades.append(round(grade / 5) * 5)
        else:
            final_grades.append(grade)
    return final_grades


print(grading_students(grades))

File: gradingStudents.js
/*
A teacher rounds students grades to the next divisible 
of 5 if they are within 2 points. Grades 37 and below 
are not rounded.
*/

grades = [73, 67, 38, 33]

function gradingStudents(grades) {
    const finalGrades = []
    for (let i = 0; i < grades.length; i++) {
        if (grades[i] <= 37) {
            finalGrades.push(grades[i])
        }
        else if ((grades[i] + 2) % 5 === 0) {
            finalGrades.push(Math.ceil(grades[i] / 5) * 5)
        }
        else if ((grades[i] + 1) % 5 === 0) {
            finalGrades.push(Math.ceil(grades[i] / 5) * 5)
        }
        else {
            finalGrades.push(grades[i])
        }
    }
    return (finalGrades)
}

console.log(gradingStudents(grades))